src/strtree.c

```diff
@@ -290,41 +301,46 @@ static PyObject *STRtree_query(STRtreeObject *self, PyObject *args) {
         return PyArray_SimpleNew(1, dims, NPY_INTP);
     }
 
+    if (predicate_id != 0) {
+        predicate_func = get_predicate_func(predicate_id);
+        if (predicate_func == NULL) {
+            return NULL;
+        }
+    }
+
+    GEOS_INIT;
```

Comment:
@caspervdw did you try to use `GEOS_INIT_THREADS` here? 

As far as I can see, it's only `copy_kvec_to_npy` that uses Python APIs, that the calls to this can always be moved after `GEOS_FINISH_THREADS` I think. 

But if I change it to `_THREADS`, I get compilation errors.

Or is there something else I am missing that requires Python in the inner query loop? (also the GeometryObjects are put in a C vector)

Reply:
I think we should try to implement the `_THREADS` versions in more places indeed. Let's treat that as separate issues.

Reply:
Yes, separate issues/PRs for sure, but I was mainly wondering if you already tried/experienced this for the STRtree

-> https://github.com/pygeos/pygeos/issues/154